### PR TITLE
Automatic compilation of Tailwind CSS

### PR DESCRIPTION
**Pull request by labnol**
You might not need a separate task for watching Tailwind CSS since `npm run dev` will handle it automatically.

### CODE REVIEW

File: tailwind.config.cjs [hunk 1]
```diff
@@ -1,6 +1,6 @@
 /** @type {import('tailwindcss').Config} */
 module.exports = {
-  content: ['./src/**/*.{html,js,ts}', './index.html'],
+  content: ['./src/client/**/*.{html,js,ts}', './index.html'],
```
Comment (WildH0g) [inline tailwind.config.cjs:3]: This misses the `src/server` folder, why skip it?

File: index.html [hunk 2]
```diff
@@ -4,7 +4,6 @@
     <meta charset="UTF-8" />
     <meta http-equiv="X-UA-Compatible" content="IE=edge" />
     <meta name="viewport" content="width=device-width, initial-scale=1.0" />
-    <link rel="stylesheet" href="./src/client/assets/output.css" />
```
Comment (WildH0g) [inline index.html:7]: How is the CSS file included if you leave it out from HTML?

File: package.json [hunk 3]
```diff
@@ -25,7 +25,6 @@
     "env:prod": "node ./env-mgt/set-env.js prod",
     "dev": "vite",
     "build:ui": "vite build",
-    "build:css:watch": "npx tailwindcss -i ./src/client/assets/input.css -o ./src/client/assets/output.css --watch",
```
Comment (WildH0g) [inline package.json:28]: Vite does not compile Tailwind, I don't see why you believe Vite executes Tailwind. What am I missing here?

File: package.json [hunk 4]
```diff
@@ -46,14 +45,16 @@
...
     "daisyui": "^3.1.5",
     "eslint": "^8.40.0",
     "eslint-config-prettier": "^8.8.0",
     "eslint-plugin-googleappsscript": "^1.0.4",
     "husky": "^8.0.3",
     "jest": "^29.5.0",
+    "postcss": "^8.4.44",
     "prettier": "^2.8.8",
-    "tailwindcss": "^3.4.6",
+    "tailwindcss": "^3.4.10",
```
Comment (WildH0g) [inline package.json:57]: I'm all for bumping versions and keeping them up-to-date, but let's figure out all the other CSS stuff first :)

File: package.json [hunk 5]
```diff
@@ -46,14 +45,16 @@
...
     "@types/google-apps-script": "^1.0.60",
     "@types/jest": "^29.5.1",
+    "autoprefixer": "^10.4.20",
     "daisyui": "^3.1.5",
     "eslint": "^8.40.0",
     "eslint-config-prettier": "^8.8.0",
     "eslint-plugin-googleappsscript": "^1.0.4",
     "husky": "^8.0.3",
     "jest": "^29.5.0",
+    "postcss": "^8.4.44",
```
Comment (WildH0g) [inline package.json]: What's this for?

File: package.json [hunk 6]
```diff
@@ -46,14 +45,16 @@
     "@types/alpinejs": "^3.13.10",
     "@types/google-apps-script": "^1.0.60",
     "@types/jest": "^29.5.1",
+    "autoprefixer": "^10.4.20",
```
Comment (WildH0g) [inline package.json:48]: What does this do?

File: postcss.config.cjs [hunk 7]
```diff
@@ -0,0 +1,6 @@
+module.exports = {
+  plugins: {
+    tailwindcss: {},
```
Comment (WildH0g) [inline postcss.config.cjs:3]: So is Post CSS handling the compilation of Tailwind?

**Review by WildH0g — COMMENTED** [review]
Hi, 

Thanks for the contribution. Can you explain in more detail the new approach? I think I got the gist of it reviewing your code, and it looks like it can improve the developer experience, but I don't fully get it.

File: tailwind.config.cjs (hunk 1 above)
Comment (labnol) [reply tailwind.config.cjs:3]: Won't the client folder hold all Tailwind CSS classes?

File: postcss.config.cjs (hunk 7 above)
Comment (labnol) [reply postcss.config.cjs:3]: This would take care of automatic compilation.

File: package.json (hunk 3 above)
Comment (labnol) [reply package.json:28]: The CSS processing happens automatically as part of your normal development and build processes.

File: index.html (hunk 2 above)
Comment (labnol) [reply index.html:7]: It will be inlined into the output since you are using the singlefile vite plugin.

File: tailwind.config.cjs (hunk 1 above)
Comment (WildH0g) [reply tailwind.config.cjs:3]: Oh right, it's tailwind. Good point.

File: index.html (hunk 2 above)
Comment (WildH0g) [reply index.html:7]: OK

File: package.json (hunk 3 above)
Comment (WildH0g) [reply package.json:28]: How exactly? `npx tailwind` watches for changes in the files, identifies the classes you are using, takes the settings from the `input` file and compiles the `output` file. If it's not running, how is the CSS processing happening?

File: postcss.config.cjs (hunk 7 above)
Comment (WildH0g) [reply postcss.config.cjs:3]: I'm going to give this a try, looks promising!

**Comment by WildH0g** [comment]
Hi @labnol ,

I've tested your code, and it's working great! I'm ready to merge the pull request. Before I do that, could you please:

- Update the version to 1.3.0 in `package.json`
- Update the version in the shield in the `README.md` (right under the heading)
- Reflect the new changes in `README.md`
- Update `HISTORY.md`

Thanks again for your contribution!

**Review by labnol — COMMENTED** [review]
Updated dependencies.

File: .gitignore [hunk 8]
```diff
@@ -8,6 +8,7 @@ pnpm-debug.log*
 lerna-debug.log*
 
 node_modules
+dist/*
```
Comment (WildH0g) [inline .gitignore]: There is a copy and paste version in the `dist` folder, probably should leave that intact

File: HISTORY.md [hunk 9]
```diff
@@ -1,5 +1,13 @@
 # History
 
+## Version 1.3.0 | 2024-08-22
```
Comment (WildH0g) [inline HISTORY.md]: Change the date please :)

**Review by WildH0g — CHANGES_REQUESTED** [review]
Just a couple of small things.

File: .gitignore (hunk 8 above)
Comment (labnol) [reply .gitignore]: Fixed.

**Comment by labnol** [comment]
Ready to merge!

**Review by labnol — COMMENTED** [review]
Updated history.md and gitignore.

**Comment by WildH0g** [comment]
Merged!

**Pull request merged**